### DISCUSSION TITLE
ESP32-8048S070 support

### DISCUSSION THREAD

**Discussion by themagicm** [Bodmer/TFT_eSPI#2381] [Compatible displays and setup files]
Do you have the required settings for the ESP32-8048S070?

**Comment by Bodmer** [comment]
Provide web links to the vendors page so I can look at the technical details.

**Comment by themagicm** [comment]
NICE!   Thank you for taking the time to look, even if it doesnt get implemented.  Appreciate that!!

This is the display itself:
https://www.makerfabs.com/sunton-esp32-s3-7-inch-tn-display-with-touch.html

..but further down on that page under "Documents" is a link to a zip file which has tech specs.   Hopefully the link below works for you, if not, its on the page towards the bottom.

http://www.jczn1688.com/zlxz?spm=a2g0o.detail.1000023.1.16c569f0Ex1SqQ

You'll want to check out this file:
7.0inch_ESP32-8048S070.zip    115megs in size

**Reply by Bodmer** [reply]
Unfortunately the answer is no, it is not suported and it is unlikely to be supported in the future. This seems to be a bespoke display that uses the processor to continuously refresh the display. This is achieved with the support of some bepoke Espressif code that runs using a full screen buffer. All rather messy to interface with unless you can read Chinese!

**Reply by themagicm** [reply]
Do you have a recommendation for a 7" display that would work with your lib and an ESP32-WROOM-32S or a Teensy 4.1?

**Reply by Bodmer** [reply]
The only 7" screen I have used is this one:
https://www.buydisplay.com/7-tft-screen-touch-lcd-display-module-w-ssd1963-controller-board-mcu

This is with an 8 bit parallel (8080) interface and I had the capacitance touch option.

The library is really intended for low cost displays, once the cost goes over $25 then it is cheaper to use an OTS packaged Android tablet!

**Reply by themagicm** [reply]
With er3304-1 font chip?  Reason for the esp/teensy is because I have software written for it that is working...and with the 8080 parallel header?

**Reply by elgerg** [reply]
Re https://github.com/Bodmer/TFT_eSPI/discussions/2381#discussioncomment-4897107

I also have the same display and would love support. I'm sure people in future would also appreciate it.

Thanks

**Reply by Bodmer** [reply]
@themagicm You do not need the font chip.

The TFT_eSPI library does not provide parallel interface support for generic processors, this includes the Teensy... the SSD1963 does not support a SPI interface, it is only parallel.

**Reply by themagicm** [reply]
Ok.  What do you recommend as a MCU to interface with that display you posted a link on that can work with your library?  (Sorry..new to this.  Been using a Nextion all this time)

**Reply by Bodmer** [reply]
ESP32, ESP32 S3, RPi Pico RP2040 all have 8 bit parallel support for writing to the display. STM32F processors can also be used.

**Reply by themagicm** [reply]
ok, so the link you posted above will not meet the criteria that I posted...works with an ESP32 and works with your lib, correct?

**Reply by Bodmer** [reply]
This screen:
https://www.buydisplay.com/7-tft-screen-touch-lcd-display-module-w-ssd1963-controller-board-mcu

Can be used with ESP32, ESP32 S3, RPi Pico RP2040 since all have 8 bit parallel support in the TFT_eSPI library for writing to the display. STM32F processors can also be used.

**Reply by themagicm** [reply]
Would it help if you looked at Arduino_GFX library since it supports ESP32-8040S070?   Your lib has nicer features which is why I'm hoping you reconsider. lol

**Comment by JuanBtaB** [comment]
This GFX settings works fine for this screen
```
Arduino_ESP32RGBPanel *rgbpanel = new Arduino_ESP32RGBPanel(
  41, 40, 39, 42,      // DE, VSYNC, HSYNC, PCLK
  14, 21, 47, 48, 45,  // R0-R4
  9, 46, 3, 8, 16, 1,  // G0-G5
  15, 7, 6, 5, 4,      // B0-B4
  0, 210, 4, 43,       // hsync // wrong: 0, 180, 30, 16, // good: 0, 210, 4, 43,
  0, 22, 4, 12,        // vsync // wrong: 0, 12, 13, 10,  // good:  0, 22, 4, 12,
  1, 16000000, false,  // active neg, speed 12000000 => Max= 16000000, useBigEndian
  0, 0, 0);            // de idle high, pclk idle high, bounce buffer size 
Arduino_RGB_Display *gfx = new Arduino_RGB_Display(800, 480, rgbpanel);
```

```
#define TOUCH_SDA 19
#define TOUCH_SCL 20
#define TOUCH_INT 18
#define TOUCH_RST 38

TAMC_GT911 ts = TAMC_GT911(TOUCH_SDA, TOUCH_SCL, TOUCH_INT, TOUCH_RST, 800, 480);

```
I prefer to use TFT_eSPI to GFX but I don't know how to make the right driver.
